plugins/Model_DFaker/Model.py

```diff
@@ -0,0 +1,172 @@
+from keras.models import Model as KerasModel
+from keras.layers import Input, Dense, Flatten, Reshape, Dropout, Add,Concatenate, Lambda
+from keras.layers.advanced_activations import LeakyReLU
+from keras.layers.convolutional import Conv2D
+from keras.initializers import RandomNormal
+from keras.optimizers import Adam
+
+from .AutoEncoder import AutoEncoder
+from lib.PixelShuffler import PixelShuffler
+
+import tensorflow as tf
+
+from keras_contrib.losses import DSSIMObjective
+from keras import losses
+
+from keras.utils import multi_gpu_model
+
+
+class penalized_loss(object):
+
+  def __init__(self,mask,lossFunc,maskProp= 1.0):
+    self.mask = mask
+    self.lossFunc=lossFunc
+    self.maskProp = maskProp
+    self.maskaskinvProp = 1-maskProp  
+
+  def __call__(self,y_true, y_pred):
+
+    tro, tgo, tbo = tf.split(y_true,3, 3 )
+    pro, pgo, pbo = tf.split(y_pred,3, 3 )
+
+    tr = tro
+    tg = tgo
+    tb = tbo
+
+    pr = pro
+    pg = pgo
+    pb = pbo
+    m  = self.mask 
+
+    m   = m*self.maskProp
+    m  += self.maskaskinvProp
+    tr *= m
+    tg *= m
+    tb *= m
+
+    pr *= m
+    pg *= m
+    pb *= m
+
+
+    y = tf.concat([tr, tg, tb],3)
+    p = tf.concat([pr, pg, pb],3)
+
+    #yo = tf.stack([tro,tgo,tbo],3)
+    #po = tf.stack([pro,pgo,pbo],3)
+
+    return self.lossFunc(y,p)
+
+IMAGE_SHAPE = (64,64,3)
+
+ENCODER_DIM = 1024
+conv_init = RandomNormal(0, 0.02)
+gamma_init = RandomNormal(1., 0.02)
+
+def __conv_init(a):
+    print("conv_init", a)
+    k = RandomNormal(0, 0.02)(a) # for convolution kernel
+    k.conv_weight = True    
+    return k
+
+class Model(AutoEncoder):
+    def initModel(self):
+        optimizer = Adam( lr=5e-5, beta_1=0.5, beta_2=0.999 )
+
+        # print(self.encoder.summary()) 
+        # print(self.decoder_A.summary())
+        # print(self.decoder_B.summary())
+
+        x1 = Input( shape=IMAGE_SHAPE )
+        x2 = Input( shape=IMAGE_SHAPE )
+        m1 = Input( shape=(64*2,64*2,1) )
+        m2 = Input( shape=(64*2,64*2,1) )
+
+        self.autoencoder_A = KerasModel( [x1,m1], self.decoder_A( self.encoder(x1) ) )
+        self.autoencoder_B = KerasModel( [x2,m2], self.decoder_B( self.encoder(x2) ) )
+
+        # print(self.autoencoder_A.summary())
+        # print(self.autoencoder_B.summary())
+
+        if self.gpus > 1:
+            self.autoencoder_A = multi_gpu_model( self.autoencoder_A ,self.gpus)
+            self.autoencoder_B = multi_gpu_model( self.autoencoder_B ,self.gpus)
+
+        # o1,om1  = self.decoder_A( self.encoder(x1))
+        # o2,om2  = self.decoder_B( self.encoder(x2))
+
+        DSSIM = DSSIMObjective()
+        self.autoencoder_A.compile( optimizer=optimizer, loss=[ penalized_loss(m1, DSSIM),'mse'] )
+        self.autoencoder_B.compile( optimizer=optimizer, loss=[ penalized_loss(m2, DSSIM),'mse'] )
+
+    def converter(self, swap):
+        autoencoder = self.autoencoder_B if not swap else self.autoencoder_A 
+        return lambda img: autoencoder.predict(img)
```

Comment:
Only passes a single parameter to the predict function, it should take two, the image and an empty mask like `zmask = numpy.zeros((1,128, 128,1),float)`

Reply:
Hmm, it is a copy/paste from other plugins, and it should be cleaned. 

However in Convert_DFaker, predict is called with `.predict([img, mask])`, so the behavior should be good.

What do you prefer? Keeping it this way, having 2 params for the lambda, or keeping & param and creating the empty mask in this function? Maybe the 3d option is best because it would keep the same behavior as other plugins...